sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -329,7 +320,7 @@ private CompressedSource(
    * CompressedSource#createForSubrangeOfFile}.
    */
   private CompressedSource(FileBasedSource<T> sourceDelegate,
-      DecompressingChannelFactory channelFactory, String filePatternOrSpec, long minBundleSize,
+      DecompressingChannelFactory channelFactory, Metadata filePatternOrSpec, long minBundleSize,
```

Comment:
Should all the parameters of type Metadata be renamed? (they're currently variously named fileOrPatternSpec, fileName etc.) - fine to ignore if it's too laborious.